examples/23_flag_subcommands_pacman.rs

```diff
@@ -0,0 +1,155 @@
+// Working with flag subcommands allows behavior similar to the popular Archlinux package manager Pacman.
+// Man page: https://jlk.fjfi.cvut.cz/arch/manpages/man/pacman.8
+//
+// It's suggested that you read examples/20_subcommands.rs prior to learning about `FlagSubCommand`s
+//
+// This differs from normal subcommands because it allows passing subcommands in the same fashion as `clap::Arg` in short or long args.
+//
+//            Top Level App (pacman)                              TOP
+//                           |
+//    ---------------------------------------------------
+//   /     |        |        |         |        \        \
+// sync  database remove    files      query   deptest   upgrade  LEVEL 1
+//
+// Given the above hierachy, valid runtime uses would be (not an all inclusive list):
+//
+// $ pacman -Ss
+//           ^--- subcommand followed by an arg in its scope.
+//
+// $ pacman -Qs
+//
+// $ pacman -Rns
+//
+// NOTE: Subcommands short flags can be uppercase or lowercase.
+//
+// $ pacman --sync --search
+//            ^--- subcommand
+//
+// $ pacman sync -s
+//          ^--- subcommand
+//
+// NOTE: this isn't valid for pacman, but is done implicitly by Clap which
+// adds support for both flags and standard subcommands out of the box.
+// Allowing your users to make the choice of what feels more intuitive for them.
+//
+// Notice only one command per "level" may be used. You could not, for example, do:
+//
+// $ pacman -SQR
```

Comment:
Oh no, it handles that if they were subcommands of one another! But they're all on the same level. :+1:

That'd be like running

```sh
git add * push
```

Reply:
What does "all on the same level" mean? I believe subcommands in clap are mutually exclusive, using seeral subcommands is only allowed if the subs are nested (latter subcommands are sub-subcommands). For example, given the CLI: 

``` 
SUBCOMMANDS:
  add
  push
```

And this invocation:
```
app add push
```

This would result in an error because `push` would be parsed and considered a subcommand of another subcommand (`add`); it's not two subcommands called sequentially.

Reply:
Yeah, so how the graph above illustrates the subcommands `S`, `Q`, and `R` are akin to `add`, `push`, and `pull` for git. Using them in the manner of `pacman -SQR` is an error, because `Q` would be treated as a subcommand or flag of `S`, when it isn't... same applies for `Q` and `R`. This is the behavior we both expect, they're treated mutually exclusive.

How should I clarify this in the documentation?

Reply:
I think I would prefer to fallback to good old prose instead of ASCII art. The graph above gives me strong impression that the subcommands are all "on one level" when in fact they are subcommands of subcommands of subcommands.

Also, I don't really like that you reiterate "each subcommand has its own namespace...". I feel strongly that this part belongs to normal subcommands documentation, and I think we can assume that the reader is familiar with normal subcommands.

With all that in mind, I propose this as the working version  
~~~
This feature allows users of the app to pass subcommands in the fashion of short or long flags. 
You may be familiar with it if you ever used [`pacman`](https://wiki.archlinux.org/index.php/pacman).

Some made up examples of what flag subcommands are:

```shell
$ pacman -S
          ^--- short flag subcommand.

$ pacman --sync
          ^--- long flag subcommand.

$ pacman -Ss
          ^--- short flag subcommand followed by a short flag
                  (users can "stack" short subcommands with short flags)

$ pacman -S -s
          ^--- same as above

$ pacman -S --sync
          ^--- short flag subcommand followed by a long flag

$ pacman -SQ
          ^--- short flag subcommand followed by a short flag subcommand
                  (users can "stack" short subcommands with each other)
                  Notice that `-S` here is a top-level subcommand, and `-Q` is sub-subcommand
```

NOTE: Keep in mind that subcommands, flags, and long flags are *case sensitive*: `-Q` and `-q` are different flags/subcommands. For example, you can have both `-Q` subcommand and `-q` flag, and they will be properly disambiguated. 

Let's make a quick program to illustrate.
~~~

And the code. Neat?